src/docs/arch-data-file.dox

```diff
@@ -1,17 +1,90 @@
+contents are not corrupted.
+
+Format & Layout of On-Disk Pages
+
+Pages consist of a header (\c WT_PAGE_HEADER and \c WT_BLOCK_HEADER) followed 
+by a variable number of cells, which encode keys, values or addresses (see 
+\c cell.h). The page header \c WT_PAGE_HEADER consists of information such 
+as the column-store record number, write generation (required for ordering 
+pages in time), in-memory size, cell count (the number of cells on the page), 
+data length (the overflow record length), and the page type. This is 
+immediately followed by the block header \c WT_BLOCK_HEADER which contains 
+block-manager specific information such as flags, version and unused.
+
+After the page header, pages with variable-length keys or values will have 
+a set of cells. There are four main types of cells - keys, data, deleted 
+cells, and off-page references. Cells are also usually followed by some 
+additional data depending on its type. For example, values may have a time 
+validity window, and off-page references may have a validity window and an 
+address cookie. Ordering is important, values are all 
+associated with the key that precedes them. Multiple values may be present, 
+which can represent multiple versions. Extremely large values may be 
+represented as a reference to another page (see overflow pages).
+
+The exact encoding of cells is rather complex, and beyond what can be 
+described here. The encoding strikes a balance between data that can be 
+compacted efficiently in time and space, extensibility, and compatibility 
+with previous versions.
+
+Page Types
+Different types of pages are made up of different types of cells.
+
+(1) Internal Pages - as the B-Tree grows in size and layer/s are added, 
+there will be pages between the root page and leaf page. These internal 
+pages contain keys that point to other pages.
+- The row-store internal page \c WT_PAGE_ROW_INT is made up of a key cell 
+(\c WT_CELL_KEY or \c WT_CELL_KEY_OVFL) followed by an offpage-reference 
+(\c WT_CELL_ADDR_XXX).
+- The column-store internal page simply contains an off-page reference.
+
+(2) Leaf Pages - leaf pages consist of a page header, followed by keys, 
+values or addresses (these reference overflow pages).
+- The row-store leaf page \c WT_PAGE_ROW_LEAF is made up of a key cell 
+(\c WT_CELL_KEY or \c WT_CELL_KEY_OVFL) followed by a value 
+(\c WT_CELL_VALUE/VALUE_COPY/VALUE_OVFL).
+- The column-store leaf page (for variable-length data) WT_PAGE_COL_VAR 
+is made up of data cells (WT_CELL_VALUE/VALUE_COPY/VALUE_OVFL) or 
+deleted cells (WT_CELL_DEL)
+
+(3) Overflow Pages - overflow pages are needed when keys/values that are 
+too large must be stored separately in the file, apart from where the item 
+is logically placed. Page sizes and configuration values such as 
+internal_key_max, leaf_key_max and leaf_value_max are used to determine 
+overflow items.
+
+Whilst the above gives a general overview of how different page types are 
```

Comment:
Whilst -> while  (whilst is archaic in US English).  I think this appears in a few spots.

Reply:
Thanks for the heads up, I didn't know this!